File: example_services/java/finatra/example/example-service/src/main/java/com/me/example/service/ExampleServer.java
package com.me.example.service;

import com.me.example.controllers.ExampleController;
import com.me.example.modules.MessageDaoModule;
import com.twitter.finatra.http.AbstractHttpServer;
import com.twitter.finatra.http.filters.CommonFilters;
import com.twitter.finatra.http.filters.LoggingMDCFilter;
import com.twitter.finatra.http.filters.TraceIdMDCFilter;
import com.twitter.finatra.http.routing.HttpRouter;

import javax.inject.Singleton;

@Singleton
class ExampleServer extends AbstractHttpServer {

    public ExampleServer() {
        addFrameworkModule(new MessageDaoModule());
    }

    @Override
    public void configureHttp(HttpRouter router) {
        router
            .filter(new LoggingMDCFilter<>())
            .filter(new TraceIdMDCFilter<>())
            .filter(CommonFilters.class)
            .add(ExampleController.class);
    }
}

File: example_services/scala/finatra/example/example-service/src/main/scala/com/me/example/service/ExampleServer.scala
package com.me.example.service

import com.me.example.controllers.ExampleController
import com.me.example.filters.ExampleAuthFilter
import com.netflix.config.{ConfigurationBasedDeploymentContext, ConfigurationManager}
import com.twitter.finagle.http.{Request, Response}
import com.twitter.finatra.http.HttpServer
import com.twitter.finatra.http.filters.{CommonFilters, LoggingMDCFilter, TraceIdMDCFilter}
import com.twitter.finatra.http.routing.HttpRouter
import com.me.example.modules.MessageDaoModule
import com.twitter.inject.Logging

object ExampleServerMain extends ExampleServer

class ExampleServer extends HttpServer with Logging {

  ConfigurationManager.loadCascadedPropertiesFromResources("example")

  info(s"DeploymentCtx=${ConfigurationManager.getDeploymentContext.getApplicationId}")
  info(s"DeploymentCtx=${ConfigurationManager.getDeploymentContext.getDeploymentEnvironment}")
  info(s"DeploymentCtx=${ConfigurationManager.getDeploymentContext.getDeploymentDatacenter}")
  info(s"ConfigurationManager Installed=${ConfigurationManager.isConfigurationInstalled}")

  // this is how you turn off the admin services
  //override val disableAdminHttpServer = true

  // add some Guice Modules
  override def modules = Seq(MessageDaoModule)

  override def configureHttp(router: HttpRouter) {
    router
      .filter[LoggingMDCFilter[Request, Response]]
      .filter[TraceIdMDCFilter[Request, Response]]
      .filter[CommonFilters]
      .filter[ExampleAuthFilter[Request]]
      .add[ExampleController]
  }
}